sources/api/sundog/src/main.rs

```diff
@@ -221,6 +224,54 @@ where
     Ok(populated_settings)
 }
 
+// Builds the proxy environment variables to pass to settings generators
+async fn build_proxy_env<P>(socket_path: P) -> Result<HashMap<String, String>>
+where
+    P: AsRef<Path>,
+{
+    // Retrieve network proxy-related settings.
+    let settings: serde_json::Value =
+        schnauzer::get_json(&socket_path, "/settings", Some(("prefix", "network")))
+            .await
+            .context(error::RetrieveSettingsSnafu)?;
+    let mut proxy_envs = HashMap::new();
+    if let Some(network_settings) = settings.get("network") {
```

Comment:
nit: I'd check for `None` and do an early return to avoid excessive indents and the related cognitive burden of keeping track of nested conditionals.